### PR TITLE
[Clustering](ft): Add clustering

### PR DESCRIPTION
**Pull request by alexandre-merle**
Add a clustering, simple to use, with an internal
watchdog

### CODE REVIEW

**Comment by alexandre-merle** [comment]
Updated :) @MathieuCassagne Did you see any feature you will need in DBD ? @LaurenSpiegel same question :)

File: README.md [hunk 1]
```diff
@@ -46,3 +46,24 @@ console.log(errors.AccessDenied);
 //    AccessDenied: true }
 
 ```
+
+## Clustering
+
+The clustering class can be used to set up a cluster of worker. The class will
```
Comment (LaurenSpiegel) [inline README.md]: of workers

File: README.md [hunk 2]
```diff
@@ -46,3 +46,24 @@ console.log(errors.AccessDenied);
...
+create at least 1 worker, will log any worker event (started, exited).
+The class has also a watchdog who will restart workers until the stop() method
+was called.
```
Comment (LaurenSpiegel) [inline README.md]: is called

File: lib/Clustering.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     * Constructor
+     *
+     * @param {number} size Cluster size
+     * @param {Logger} logger Logger object
+     * @return {Clustering} itself
+     */
+    constructor(size, logger) {
+        this._size = size;
+        if (size < 1) {
+            throw new Error('Cluster size must superior or equal to 1');
```
Comment (LaurenSpiegel) [inline lib/Clustering.js]: must be greater than or equal to

**Comment by LaurenSpiegel** [comment]
We're not listening to the on "disconnect" event here.  I need some convincing that this is as safe as the setInterval approach in S3's current server.

**Pull request closed without merging**